lib/redux/reducers/uploads.js

```diff
@@ -121,16 +122,33 @@ export function uploadsByUser(state = {}, action) {
         });
       });
       if (uploadTargetUser && uploadTargetDevice) {
-        return update(
-          state,
-          {[uploadTargetUser]: {[uploadTargetDevice]: {
-            choosingFile: {$set: false},
-            completed: {$set: true},
-            error: {$set: err},
-            failed: {$set: true}
-          }}}
-        );
+        let devicesForCurrentUser = _.get(state, [uploadTargetUser], {});
```

Comment:
Nitpick: I find it a bit weird to use the array syntax for `_.get` with just one value. And about 70% certain that the rest of the codebase here in the uploader doesn't use the array syntax in these situations, but not 100% certain...